lib/client.dart

```diff
@@ -1,59 +1,128 @@
 import 'dart:async';
 import 'dart:convert';
+
+import 'package:gql/execution.dart' as exec;
+import 'package:gql/language.dart';
+import 'package:gql/link.dart' as gql_link;
 import 'package:http/http.dart' as http;
 import 'package:json_annotation/json_annotation.dart';
+
 import './schema/graphql_error.dart';
 import './schema/graphql_query.dart';
 
-class _DefaultHttpJsonClient extends http.BaseClient {
+/// Simple implementation of an HttpLink
+class SimpleHttpJsonLink implements gql_link.Link {
+  /// Endpoint of a GraphQL service
+  final String graphQLEndpoint;
+
   http.Client _httpClient = http.Client();
 
-  _DefaultHttpJsonClient();
+  /// Construct the Link
+  SimpleHttpJsonLink(this.graphQLEndpoint);
+
+  Future<exec.Response> _post(exec.Request request) async {
+    final dataResponse = await _httpClient.post(
+      graphQLEndpoint,
+      headers: {
+        'Content-type': 'application/json',
+        'Accept': 'application/json',
+      },
+      body: json.encode({
+        'operationName': request.operation.operationName,
+        'variables': request.operation.variables,
+        'query': printNode(request.operation.document),
+      }),
+    );
+
+    final Map<String, dynamic> jsonBody =
+        json.decode(utf8.decode(dataResponse.bodyBytes));
+
+    final response = GraphQLResponse.fromJson(jsonBody);
+
+    // TODO: implement some `GraphQLJsonResponse`
+    // to avoid this re-mapping the response.
+    //
+    // `package:gql:execution` does not care about the
+    // transport protocol and the format. It is `SimpleHttpJsonLink`s
+    // responsibility because it chooses to use Http and Json.
+    return exec.Response(
+      errors: response.errors?.map(
+        (e) => exec.GraphQLError(
+          message: e.message,
+          locations: e.locations?.map(
+            (l) => exec.ErrorLocation(
+              line: l.line,
+              column: l.column,
+            ),
+          ),
+          path: e.queryPath,
+        ),
+      ),
+      data: jsonBody['data'],
+    );
+  }
 
   @override
-  Future<http.StreamedResponse> send(http.BaseRequest request) {
-    final jsonHeaders = {
-      'Content-type': 'application/json',
-      'Accept': 'application/json',
-    };
-    request.headers.addAll(jsonHeaders);
-    return _httpClient.send(request);
+  Stream<exec.Response> request(exec.Request request,
```

Comment:
Why is it a Stream? Would `Link` work with subscriptions?

Reply:
Yes, the intention is that `Link` works with any kind of operations.